client/src/main/python/slipstream/cloudconnectors/BaseCloudConnector.py

```diff
@@ -866,3 +866,69 @@ def _has_vm_failed(self, vm_instance):
         Returns: True or False."""
         vm_state = self._vm_get_state(vm_instance).lower()
         return vm_state in [fstate.lower() for fstate in self._get_vm_failed_states()]
+
+    def resize(self, node_instances, done_reporter=None):
+        self._scale_action_runner(self._resize_and_report, node_instances, done_reporter)
+
+    # TODO: use decorator for reporter.
+    def _resize_and_report(self, node_instance, reporter):
+        self._resize(node_instance)
+        reporter(node_instance)
+
+    def attach_disk(self, node_instances, done_reporter=None):
+        self._scale_action_runner(self._attach_disk_and_report, node_instances, done_reporter)
+
+    def _attach_disk_and_report(self, node_instance, reporter):
+        self._attach_disk(node_instance)
+        reporter(node_instance)
+
+    def detach_disk(self, node_instances, done_reporter=None):
+        self._scale_action_runner(self._detach_disk_and_report, node_instances, done_reporter)
+
+    def _detach_disk_and_report(self, node_instance, reporter):
+        self._detach_disk(node_instance)
+        reporter(node_instance)
+
+    def _scale_action_runner(self, scale_action, node_instances, done_reporter):
+        """
+        :param scale_action: task executor
+        :type scale_action: callable
+        :param node_instances: list of node instances
+        :type node_instances: list [NodeInstance, ]
+        :param done_reporter: function that reports back to SlipStream
+        :type done_reporter: callable with signature `done_reporter(<NoneInstance>)`
+        """
+        max_workers = self._get_max_workers(self.configHolder)
+        scaler = VmScaler(scale_action, max_workers, self.verboseLevel)
+        scaler.set_tasks_and_run(node_instances, done_reporter)
+        scaler.wait_tasks_finished()
+
+
+class VmScaler(object):
```

Comment:
Why not extracting it into an new file (since BaseCloudConnector is already huge) ?